packages/next/src/server/base-server.ts

```diff
@@ -2769,6 +2771,23 @@ export default abstract class Server<ServerOptions extends Options = Options> {
         )
       }
 
+      if (cachedData.headers) {
+        const resHeaders = cachedData.headers
+        for (const key of Object.keys(resHeaders)) {
+          if (key === NEXT_CACHE_TAGS_HEADER) {
+            // Not sure if needs to be special.
```

Comment:
This check is trying to mimic this behavior:

https://github.com/vercel/next.js/blob/canary/packages/next/src/server/base-server.ts#L2750-L2752